worlds/yoshisisland/Options.py

```diff
@@ -0,0 +1,276 @@
+from typing import Dict, Union
+from BaseClasses import MultiWorld
+from Options import Toggle, DefaultOnToggle, DeathLink, Choice, Range, Option, OptionDict
+from schema import Schema, And, Optional
+
+class ExtrasEnabled(Toggle):
+    """If enabled, the more difficult Extra stages will be added into logic. Otherwise, they will be inaccessible."""
+    display_name = "Include Extra Stages"
+
+class SplitExtras(Toggle):
+    """If enabled, Extra stages will be unlocked individually. Otherwise, there will be a single 'Extra Panels' item that unlocks all of them."""
+    display_name = "Split Extra Stages"
+
+class SplitBonus(Toggle):
+    """If enabled, Bonus Games will be unlocked individually. Otherwise, there will be a single 'Bonus Panels' item that unlocks all of them."""
+    display_name = "Split Bonus Games"
+
+class ObjectVis(Choice):
+    """This will determine the default visibility of objects revealed by the Magnifying Glass.
+    Strict Logic will expect the Secret Lens or a Magnifying Glass to interact with hidden clouds if they are not set to visible by default."""
+    display_name = "Hidden Object Visibility"
+    option_none = 0
+    option_coins_only = 1
+    option_clouds_only = 2
+    option_full = 3
+    default = 1
+
+class SoftlockPrevention(DefaultOnToggle):
+    """If enabled, holding R + X to warp to the last used Middle Ring, or the start of the level if none have been activated."""

```

Comment:
Yoshi's Island does not have a timer function. If the player gets softlocked, by default, their only option is to start-select out and reset the entire level, which, from experience, can be frustrating. This code mimics the function of a death, to allow warping back to a checkpoint instead. The reason it's an option was because of purists, but, I believe also as a sort of accessibility feature? This way, people _know_ that it's on, and it also gives a convenient place for me to put an explanation because people would probably miss the explanation if I put it on the info page. ...Maybe.